### PR TITLE
Fix for 'Show Solutions'. Correct answers are now displayed, and chec…

### PR DESCRIPTION
**Pull request by NVieregg**
Show Solutions fixed in SCMC. Please check if:

- [x] Correct answers are checked, wrong answers are crossed
- [x] All correct answers are displayed below.

Note: we should consider renaming the SCMC Block. Discuss in tomorrows meeting.

### CODE REVIEW

File: src/plugins/laya-la-scmc/view.vue [hunk 1]
```diff
@@ -249,7 +249,7 @@ export default {
      */
```
Comment (cmauf) [inline src/plugins/laya-la-scmc/view.vue:249]: Please update the "Last Updated" Line in function comment (also for `diffSolution`)

File: src/plugins/laya-la-scmc/view.vue [hunk 2]
```diff
@@ -249,7 +249,7 @@ export default {
...
     populateShowSolutions () {
       for (let i = 0; i < this.options.length; i++) {
-        if (this.solutions[i] === null || this.solutions[i] === false) {
+        if (this.solutions.includes(i)) {
           this.showSolutions.push(this.options[i].text)
```
Comment (cmauf) [inline src/plugins/laya-la-scmc/view.vue:253]: `showSolutions` is not returned in preview

File: src/plugins/laya-la-scmc/view.vue [hunk 3]
```diff
@@ -312,16 +312,16 @@ export default {
...
         } else {
           // but is answer ?
           if (this.answers.includes(i)) {
             this.eval[i] = { 'far fa-times-circle text-danger': true }
-          } else {
-            this.eval[i] = { 'far fa-check-circle text-success': true }
-          }
+          }// else {
+          // this.eval[i] = { 'far fa-check-circle text-success': true }
+          // }
```
Comment (cmauf) [inline src/plugins/laya-la-scmc/view.vue:324]: these can be simplified to only contain l.314

File: src/plugins/laya-la-scmc/view.vue (hunk 2 above)
Comment (cmauf) [inline src/plugins/laya-la-scmc/view.vue:253]: This doesn't use the simple language alternative when `courseSimple` is true

**Review by cmauf — CHANGES_REQUESTED** [review]
Some issues in the code (see comments)

Furthermore, I think we should enforce having only one possible answer for Multiple Choice questions by using radio buttons instead of check boxes.

**Comment by NVieregg** [comment]
> Some issues in the code (see comments)
> 
> Furthermore, I think we should enforce having only one possible answer for Multiple Choice questions by using radio buttons instead of check boxes.

I just checked, and the buttons are already of the type 'radio'?

File: src/plugins/laya-la-scmc/view.vue [hunk 4]
```diff
@@ -303,25 +305,16 @@ export default {
...
-            this.eval[i] = { 'far fa-times-circle text-danger': true }
-          }// else {
-          // this.eval[i] = { 'far fa-check-circle text-success': true }
-          // }
+        if (this.solutions.includes(i) && this.answers.includes(i)) {
+          // is correct answer ?
+          this.eval[i] = { 'far fa-check-circle text-success': true }
+        } else if (this.answers.includes(i)) {
+          // is wrong answer ?
+          this.eval[i] = { 'far fa-times-circle text-danger': true }
```
Comment (cmauf) [inline src/plugins/laya-la-scmc/view.vue:317]: If no answer is is given, no item will be marked correct or wrong. In this case, I would suggest a cross mark at the correct answer.

**Review by cmauf — CHANGES_REQUESTED** [review]
Show icon when no answer is given

**Comment by NVieregg** [comment]
Simply changing the button type on "correct?" to radio does not work, no solutions are displayed and I don't know why.... We also need to add a conditional, since radio buttons should only be displayed for single choice? Will probably take longer than 5 minutes...

**Comment by cmauf** [comment]
@NVieregg I added some lines, please revisit then ping me

**Comment by NVieregg** [comment]
@cmauf looks really good!

**Pull request merged**